mmv1/third_party/terraform/resources/resource_sql_user.go

```diff
@@ -122,13 +134,19 @@ func resourceSqlUserCreate(d *schema.ResourceData, meta interface{}) error {
 	password := d.Get("password").(string)
 	host := d.Get("host").(string)
 	typ := d.Get("type").(string)
+	disabled := d.Get("disabled").(bool)
+	serverRoles := expandStringArray(d.Get("server_roles").(string))
```

Comment:
```suggestion
	serverRoles := expandStringArray(d.Get("server_roles"))
```

`serverRoles` is currently an `interface{}` containing a `[]interface{}` whose values are each `interface{}`s containing `string`, so a direct cast doesn't work here. That's what this expand fn does, it deeply casts the values into `string` values to transform this into a `[]string`.